ingestion/src/metadata/ingestion/source/bigquery_usage.py

```diff
@@ -130,16 +135,48 @@ def next_record(self) -> Iterable[TableQuery]:
                                 service_name=self.config.service_name,
                             )
                             yield tq
+                            self.ingest_lineage(tq.sql)
+
         except Exception as err:
             logger.error(repr(err))
 
-    def close(self):
-        pass
-
     def get_status(self) -> SourceStatus:
         return self.status
 
     def close(self):
         super().close()
         if self.temp_credentials:
             os.unlink(self.temp_credentials)
+
+    def ingest_lineage(self, sql):
+        result = LineageRunner(sql)
+        sink = MetadataRestSink(self.ctx, self.config, self.metadata_config)
```

Comment:
Hi, sorry random question. Should we be calling the `sink` from the `source`? Feels like a weird shortcut of the workflow definition.

If we want to ingest the lineage information here we might want to actually update the `next_record` call to return lineage information as well. Then, the sink that is configured in the workflow would pick that up AFAIK.

Reply:
agree with @pmbrull This doesn't look right to me

Reply:
if we yield lineage from usage pipelines, it goes to query parser which throws error as it is not configured to take the AddLineage data.
cc: @harshach @pmbrull

Reply:
oooh, I see. Then, let's add a check on the processor to validate if the incoming record is of type `TableQuery`. If it is, process and return the parsed query, otherwise return the record unmodified